cli/types/cypress.d.ts

```diff
@@ -142,6 +142,33 @@ declare namespace Cypress {
     clear: (keys?: string[]) => void
   }
 
+  // TODO: raise minimum required TypeScript version to 3.7
+  // and make this recursive (replace 'any' with Storable)
+  // https://github.com/cypress-io/cypress/issues/24875
+  type Storable =
+    | string
+    | number
+    | boolean
+    | null
+    | { [property: string]: any }
+    | any[]
+
+  type StorableRecord = Record<string, Storable>
+
+  interface OriginStorage {
+    origin: string
+    value: StorableRecord
```

Comment:
Shouldn't storable record be an object of Key value pairs where both key & value are type string?

Reply:
That's how it's stored by the browser, but we parse it into JavaScript before yielding it to the user, so the value can be anything JSON-stringifiable.